pkg/apis/proxy/v1alpha1/managedproxyserviceresolver_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&ManagedProxyServiceResolver{}, &ManagedProxyServiceResolverList{})
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster
+
+// +genclient
+// +genclient:nonNamespaced
+// ManagedProxyServiceResolver defines a target service that need to expose from a set of managed clusters to the hub.
+type ManagedProxyServiceResolver struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ManagedProxyServiceResolverSpec   `json:"spec,omitempty"`
+	Status ManagedProxyServiceResolverStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ManagedProxyServiceResolverList contains a list of ManagedProxyServiceResolver
+type ManagedProxyServiceResolverList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ManagedProxyServiceResolver `json:"items"`
+}
+
+// ManagedProxyServiceResolverSpec defines the desired state of ManagedProxyServiceResolver.
+type ManagedProxyServiceResolverSpec struct {
```

Comment:
it is not clear on how a consumer can access the service.

Reply:
Updated the description and add the usage example for consumers.